model/riscv_insts_vext_vset.sail

```diff
@@ -0,0 +1,164 @@
+/* ************************************************************************ */
+/* This file implements part of the vector extension.                       */
+/* Chapter 6: configuration setting instructions                            */
+
+/* ************************************************************************ */
+
+mapping sew_flag : string <-> bits(3) = {
+  "e8"    <-> 0b000,
+  "e16"   <-> 0b001,
+  "e32"   <-> 0b010,
+  "e64"   <-> 0b011,
+  "e128"  <-> 0b100,
+  "e256"  <-> 0b101,
+  "e512"  <-> 0b110,
+  "e1024" <-> 0b111
+} 
+
+mapping maybe_lmul_flag : string <-> bits(3) = {
+  ""              <-> 0b000, /* m1 by default */
+  sep() ^ "mf8"   <-> 0b101,
+  sep() ^ "mf4"   <-> 0b110,
+  sep() ^ "mf2"   <-> 0b111,
+  sep() ^ "m1"    <-> 0b000,
+  sep() ^ "m2"    <-> 0b001,
+  sep() ^ "m4"    <-> 0b010,
+  sep() ^ "m8"    <-> 0b011
+}
+
+mapping maybe_ta_flag : string <-> bits(1) = {
+  ""           <-> 0b0, /* tu by default */
+  sep() ^ "ta" <-> 0b1,
+  sep() ^ "tu" <-> 0b0
+}
+
+mapping maybe_ma_flag : string <-> bits(1) = {
+  ""           <-> 0b0, /* mu by default */
+  sep() ^ "ma" <-> 0b1,
+  sep() ^ "mu" <-> 0b0
+}
+
+/* ******************** vsetvli & vsetvl *********************** */
+union clause ast = VSET_TYPE : (vsetop, bits(1), bits(1), bits(3), bits(3), regidx, regidx)
+
+mapping encdec_vsetop : vsetop <-> bits(4) ={
+  VSETVLI <-> 0b0000,
+  VSETVL  <-> 0b1000
+}
+
+mapping clause encdec = VSET_TYPE(op, ma, ta, sew, lmul, rs1, rd)  
+  <-> encdec_vsetop(op) @ ma @ ta @ sew @ lmul @ rs1 @ 0b111 @ rd @ 0b1010111
+
+function clause execute VSET_TYPE(op, ma, ta, sew, lmul, rs1, rd) = {
+  AVL      : int  = undefined;
+  VLMAX    : int  = undefined;
+  SEW_new  : int  = undefined;
+  LMUL_new : real = undefined;
+
+  let vlen      : int  = get_vlen();
+  let LMUL_ori  : real = get_vtype_LMUL();
+  let SEW_ori   : int  = get_vtype_vsew();
+  let ratio_ori : real = to_real(SEW_ori) / LMUL_ori;
+  
+  /****** vsetvli part *******/
+  if (op == VSETVLI) then{
+    let vtype_reserved = zeros(sizeof(xlen) - 9);
+    let vtype_val : xlenbits = 0b0 @ vtype_reserved @ ma @ ta @ sew @ lmul;
+    writeCSR(csr_name_map("vtype"), vtype_val);
+    let LMUL_new  : real = get_vtype_LMUL();
+    let SEW_new   : int = get_vtype_vsew();
+    VLMAX = floor(LMUL_new) * vlen / SEW_new /* VLMAX = LUML * VLEN / SEW */
+  };
+
+  /****** vsetvl part *******/
```

Comment:
Should this be a switch instead? At least use else if...

Reply:
Match statements are used instead.